docs/relationship_synthesis.md

```diff
@@ -0,0 +1,252 @@
+supported relationship types list are accepted.
+
+#### Source & Target Resolvers
+
+Resolvers define how the source and target GUIDs are determined. There are three types of resolvers:
+
+1. `extractGuid`: Extracts the GUID from an existing attribute in the telemetry.
+2. `buildGuid`: Constructs the GUID using various pieces of information from the telemetry.
+3. `lookupGuid`: Performs a lookup using candidates to find the matching GUID based on telemetry attributes. 
+
+##### extractGuid
+
+The `extractGuid` resolver is the simplest and allows specifying an attribute to use as the GUID.
+
+Example:
+
+```yaml
+# ...
+  source:
+    extractGuid:
+      attribute: host.guid
+      entityType:
+        value: HOST
+  target:
+    extractGuid:
+      attribute: entityGuid
+      entityType:
+        attribute: nr.entityType
+# ...
+```
+
+If the resolved GUID corresponds to an `INFRA-NA` GUID, the relationship platform needs to know the actual type behind the GUID. 
+This can be achieved by providing a second field that specifies a hardcoded value for the entity type 
+or retrieves it from an existing attribute.
+
+##### buildGuid
+
+The `buildGuid` resolver allows constructing the GUID from different pieces of information.
+
+Example:
+
+```yaml
+# ...
+source:
+  buildGuid:
+    account:
+      attribute: accountId
+      # or lookup: yes
+    domain:
+      value: EXT
+    type:
+      value: SERVICE
+      valueInGuid: NA
+    identifier:
+      fragments:
+      - value: "k8s:"
+      - attribute: "service.name"
+      - value: ":pod:"
+      hashAlgorithm: FARM_HASH
+# ...
+```
+
+The `account` field can be defined using an attribute in the telemetry or by specifying a lookup key. 
+When a lookup key is defined, it will be searched within the accounts of the trusted account for the current data point (proposal system).
+
+The domain and type fields can be either hardcoded values or attributes. The type field also allows setting the given value as the type in the 
+GUID using `valueInGuid`, which is useful for building `INFRA-NA` GUIDs.
+
+The `identifier` can be constructed using one or multiple fragments. Each fragment can be a hardcoded string, an attribute, or a part of an attribute. 
+The identifier can be hashed using one of the supported hashing algorithms, with farmHash being the currently supported algorithm.
+
+If needed, a part of an attribute can be provided as a capture group in a regular expression, which is especially useful for MRP cases.
+
+##### lookupGuid
+The `lookupGuid` resolver is used when the GUID cannot be inferred from the telemetry and a lookup is required using candidates.
+
+Example:
+
+```yaml
+# ...
+lookupGuid:
+  candidateCategory: DATABASE
+  fields:
+    - field: endpoint
+      attribute: aws.db.host
+# ...
+```
+
+The `lookupGuid` points to the candidate category to be used and maps the fields required for the lookup to the attributes in the telemetry.
```

Comment:
we should point to candidate relationships here somehow.

Reply:
First attempt at linking both here: https://github.com/newrelic/entity-definitions/pull/1138/commits/1d8c80b254689c592e8e15d06133646239d42824